### PR TITLE
Support datadog plugin helm chart

### PR DESCRIPTION
**Pull request by bernardjkim**
This PR adds support for the Datadog Incident Management plugin helm chart.

changelog: Add support for the Datadog Incident Management plugin helm chart.

### CODE REVIEW

**Review by r0mant — APPROVED** [review]
lgtm but I'd get @hugoShaka's checkmark as well

File: examples/chart/access/datadog/values.yaml [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+  # containing the Datadog applicationKey. When this value is set, you must create the
+  # Secret before creating the chart release.
+  applicationKeyFromSecret: ""
+  # datadog.applicationKeySecretPath(string) -- is the Kubernetes Secret key
+  # containing the Datadog Application key. The secret name is set via `datadog.applicationKeyFromSecret`.
+  applicationKeySecretPath: "datadogApplicationKey"
+  # datadog.fallbackRecipient(string) -- specifies the default recipient for
+  # Access Request notifications. The recipient can be a Datadog user email or
+  # a team handle.
+  fallbackRecipient: "admin@example.com"
```
Comment (hugoShaka) [inline examples/chart/access/datadog/values.yaml:76]: Can you remove the default value and put the example email in the comment/godoc? Users will ignore this field and the plugin will try notify non-existing users.

**Review by hugoShaka — APPROVED** [review]
LGTM

If you want to walk the extra mile and improve the chart UX, the slack Helm chart has a native tbot integration that would be nice to have here. The diff looks like this https://github.com/gravitational/teleport/pull/46353/files and allows users to not have to deal with Teleport identities and connect with minimal values:

```yaml
datadog:
  apiKey: abc
  applicationKey: def
  fallbackRecipient: john@example.com

tbot:
  enabled: true
  clusterName: example.teleport.sh
  teleportProxyAddress: example.teleport.sh:443
  joinMethod: token
  token: <INSERT TOKEN HERE>
```

**Review by hugoShaka — APPROVED** [review]
Thank you for the tbot integration :)

**Comment by hugoShaka** [comment]
Should this be backported to v16?

**Comment by bernardjkim** [comment]
Yeah, we'll want to backport to v16.

**Pull request merged**